packages/strapi-plugin-users-permissions/config/schema.graphql

```diff
@@ -158,8 +173,38 @@ module.exports = {
           return {
             user,
           };
-        },
+        }
       },
-    },
-  },
+      register: {
+        description: 'Register a user',
+        plugin: 'users-permissions',
+        resolverOf: 'Auth.register',
+        resolver: async (obj, options, {context}) => {
+          context.request.body = _.toPlainObject(options.input);
+          
+          await strapi.plugins['users-permissions'].controllers.auth.register(context);
+          let output = context.body.toJSON ? context.body.toJSON() : context.body;
```

Comment:
You need to check if the output is an error and throw in that case. same in login

Reply:
@alexandrebodin  Isn't this kind of redundant? If it has errors from Auth controller this will return the error from that controller correctly with messages and error codes (confirmed this myself). These resolvers were intended only for relaying Auth controller's output so isn't dissecting the error output to throw the same thing out repetitive?

Reply:
I just added this code to my project in extensions/user-permissions/config/schema-graphql and it works very fine, I hope this will be part of the core soon.
I can confirm the missing error output as described. What I'm wondering is, why there will be now error thrown at line 185 (await strapi...). I compared it to updateUser mutation and when there is an error (e.g. not existing id) it already throws here:
```
await strapi.plugins['users-permissions'].controllers.user.update(context);
```
with `"message": "entry.notFound"` error response.
So no need to check context.body for an error. I think its best to keep it that way to stay in line. But right now I don't see where to go in order to achieve similar behaviour.

Reply:
Ok, difference is that e.g. updateUser throws an error by bookshelf (if used) when there is an invalid database query like the id doesn't exist, whereas strapi returns or sets ctx.badRequest() (or other errors, see @hapi/boom docs) which will not throw an error but return an HTTP-friendly error object. So GraphQL resolvers have to handle it by themselves?

Reply:
As saif by Kriskator, when using register or login the controller might return  a badRequest error this will not be caught but will set the ctx.body to this error. this is why you need to check for it so you can throw an ApolloError

Reply:
@alexandrebodin Would this be viable?

Reply:
ApolloErrors aren't used in whole project.
I did it this way:
```
if (output.isBoom) {
  throw new Error(output.data[0].messages[0].message);
}
```

Reply:
Oh, just saw the update, so now you can throw an ApolloError :)

Reply:
If you can just put this util in a function at the top of the file ;) thanks !

Reply:
@alexandrebodin here you go :)